crates/ra_db/src/input.rs

```diff
@@ -111,11 +112,56 @@ impl CrateGraph {
     pub fn dependencies<'a>(
         &'a self,
         crate_id: CrateId,
-    ) -> impl Iterator<Item = &'a Dependency> + 'a {
+    ) -> impl Iterator<Item=&'a Dependency> + 'a {
         self.arena[&crate_id].dependencies.iter()
     }
+    fn dfs_find(&self, target: CrateId, from: CrateId) -> bool {
+        for dep in self.dependencies(from) {
+            let crate_id = dep.crate_id();
+            if crate_id == target {
+                return true;
+            }
+            if self.arena.contains_key(&crate_id) {
```

Comment:
I think we can remove this check?

The user must `add_crate` before adding deps.